flatnav/util/ParallelConstructs.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <atomic>
+#include <cstdint>
+#include <exception>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+namespace flatnav {
+
+/**
+ * @brief template for executing a function in parallel using STL's threading
+ * library This is preferred in lieu of OpenMP only because it will not require
+ * having logic for installing OpenMP on the host system while installing the
+ * Python library.
+ *
+ * @tparam Function
+ * @param start_index
+ * @param end_index
+ * @param num_threads
+ * @param function
+ */
+template <typename Function>
+void executeInParallel(uint32_t start_index, uint32_t end_index,
```

Comment:
This is really nice, and thanks for removing the OpenMP dependency (will make python builds a bit easier).

Nit / question: I wonder if you could use variadic arguments to allow the function to take any number of parameters, of arbitrary type? Not necessary for this application, of course.

Reply:
That's a great suggestion. I think for now the function is expected to just take a single argument but variadic templates would make this even more general. I'm gonna try that.